### ISSUE TITLE
Role-specific rules not enforced for roles in namespace subdirectories

### ISSUE THREAD

**Issue by hlohmar** [ansible/ansible-lint#5079]
##### Summary

`role-name` (and likely other role-specific rules) does not trigger for roles stored under a namespace subdirectory (`roles/<namespace>/<role>/`) in any of the following invocations:

- `ansible-lint` (project auto-discovery)
- `ansible-lint roles/<namespace>/` (directly in namespace dir)

The rule only triggers when the exact role name is given: `ansible-lint roles/<namespace>/<role>/`.
This makes namespace-organized role layouts invisible to role-specific linting unless every role is specified one by one.

Possibly related: #2502

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 26.4.0 using ansible-core:2.21.1 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```

- ansible installation method: pip
- ansible-lint installation method: pip (also reproduced against current main branch)

##### STEPS TO REPRODUCE

Project structure:

```
ansible.cfg:
  [defaults]
  roles_path = ./roles/my_namespace

roles/
  my_namespace/
    myBadRoleName/
      tasks/main.yml
```

```console (paste below)
# no violation, role not discovered
ansible-lint

# no violation, namespace dir not traversed
ansible-lint roles/my_namespace/

# violation (role-name) correctly reported:
ansible-lint roles/my_namespace/myBadRoleName/

# moving the role to standard depth triggers violation correctly
mv roles/my_namespace/myBadRoleName roles/myBadRoleName
ansible-lint
```

##### Desired Behavior

Role-specific rules trigger for all roles regardless of namespace nesting depth, consistent with how `roles_path` in `ansible.cfg` is already used to locate role content.

Questions for maintainers:
- Is `roles/<namespace>/<role>/` layout a supported structure, or is only `roles/<role>/` expected?
- Should `roles_path` in `ansible.cfg` influence which parts are treated as role roots for rule evaluation?
- Is there a config option to declare additional role root paths for linting purposes?

##### Actual Behavior

`role-name` is silently skipped for namespace-nested roles in project auto-discovery mode and when the namespace directory is given as a path. Task content inside the same roles is linted normally, only role-specific rules are affected.

```console
# ansible-lint (role at roles/my_namespace/myBadRoleName/):
Passed: 0 failure(s), 0 warning(s) in 2 files processed of 16 encountered.

# ansible-lint roles/my_namespace/:
Passed: 0 failure(s), 0 warning(s) in 1 files processed of 2 encountered.

# ansible-lint roles/my_namespace/myBadRoleName/:
WARNING  Listing 1 violation(s) that are fatal
role-name: Role name myBadRoleName does not match ``^[a-z][a-z0-9_]*$`` pattern.
roles/my_namespace/myBadRoleName:1
Failed: 1 failure(s), 0 warning(s) in 2 files processed of 2 encountered.

# ansible-lint (role moved to roles/myBadRoleName/):
WARNING  Listing 1 violation(s) that are fatal
role-name: Role name myBadRoleName does not match ``^[a-z][a-z0-9_]*$`` pattern.
roles/myBadRoleName:1
Failed: 1 failure(s), 0 warning(s) in 3 files processed of 17 encountered.
```

Verified against current main branch and latest release.

**Comment by santosh7676** [comment]
Reproduced in code: _extend_with_roles() walks up to the first parent of roles/ (roles/my_namespace/) instead of the actual role directory, and the role kind glob **/roles/*/ only matches one level. Happy to work on a fix.
